packages/fluentui/react-northstar/src/components/Embed/Embed.tsx

```diff
@@ -60,143 +76,173 @@ export interface EmbedProps extends UIComponentProps {
-        className={classes.root}
-        onClick={this.handleClick}
-        {...accessibility.attributes.root}
-        {...unhandledProps}
-        {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
-      >
-        {active && (
-          <>
-            {Video.create(video, {
-              defaultProps: () => ({
+  const { classes, styles: resolvedStyles } = useStyles<EmbedStylesProps>(Embed.displayName, {
+    className: embedClassName,
+    mapPropsToStyles: () => ({
+      iframeLoaded,
+      active,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
+
+  const placeholderElement = placeholder ? (
+    <Image
+      src={placeholder}
+      styles={resolvedStyles.image}
+      variables={{ width: variables.width, height: variables.height }}
+    />
+  ) : null;
+
+  const hasIframe = !_.isNil(iframe);
+  const hasVideo = !_.isNil(video);
+  const controlVisible = !active || hasVideo;
+  const placeholderVisible = !active || (hasIframe && active && !iframeLoaded);
+
+  const element = (
+    <ElementType
+      {...getA11yProps('root', {
+        className: classes.root,
+        onClick: handleClick,
+        ...unhandledProps,
+      })}
+    >
+      {active && (
+        <>
+          {Video.create(video, {
+            defaultProps: () =>
+              getA11yProps('video', {
                 autoPlay: true,
                 controls: false,
                 loop: true,
                 muted: true,
                 poster: placeholder,
-                styles: styles.video,
+                styles: resolvedStyles.video,
                 variables: {
                   width: variables.width,
                   height: variables.height,
                 },
               }),
-            })}
-            {iframe && (
-              <Ref innerRef={this.frameRef}>
-                {Box.create(iframe, {
-                  defaultProps: () => ({
-                    as: 'iframe',
-                    styles: styles.iframe,
-                  }),
-                  overrideProps: this.handleFrameOverrides,
-                })}
-              </Ref>
-            )}
-          </>
-        )}
-
-        {placeholderVisible && placeholderElement}
-        {controlVisible &&
-          Box.create(control, {
-            defaultProps: () => ({
+          })}
+          {iframe && (
+            <Ref innerRef={frameRef}>
+              {Box.create(iframe, {
+                defaultProps: () => ({
+                  as: 'iframe',
+                  styles: resolvedStyles.iframe,
+                }),
+                overrideProps: handleFrameOverrides,
+              })}
+            </Ref>
+          )}
+        </>
+      )}
+
+      {placeholderVisible && placeholderElement}
+      {controlVisible &&
+        Box.create(control, {
+          defaultProps: () =>
+            getA11yProps('control', {
```

Comment:
Please why do you call `getA11yProps` for control slot, but not for iframe slot on line 193?